rkt/fly.go

```diff
@@ -0,0 +1,164 @@
+	"github.com/coreos/rkt/pkg/aci"
+	"github.com/coreos/rkt/store"
+)
+
+var (
+	cmdFly = &cobra.Command{
+		Use:   "fly IMAGE [ -- image-args...]",
+		Short: "Run a single application image with no pod or isolation",
+		Long:  `IMAGE should be a string referencing an image; either a hash, local file on disk, or URL.`,
+		Run:   runWrapper(runFly),
+	}
+)
+
+func init() {
+	cmdRkt.AddCommand(cmdFly)
+
+	// Disable interspersed flags to stop parsing after the first non flag
+	// argument. All the subsequent parsing will be done by parseApps.
+	// This is needed to correctly handle image args
+	cmdFly.Flags().SetInterspersed(false)
+}
+
+func runFly(cmd *cobra.Command, args []string) (exit int) {
+	err := parseApps(&rktApps, args, cmd.Flags(), true)
+	if err != nil {
+		stderr("fly: error parsing app image arguments: %v", err)
+		return 1
+	}
+
+	if rktApps.Count() != 1 {
+		stderr("fly: must provide exactly one image")
+		return 1
+	}
+
+	if globalFlags.Dir == "" {
+		log.Printf("fly: dir unset - using temporary directory")
+		var err error
+		globalFlags.Dir, err = ioutil.TempDir("", "rkt")
+		if err != nil {
+			stderr("fly: error creating temporary directory: %v", err)
+			return 1
+		}
+	}
+
+	s, err := store.NewStore(globalFlags.Dir)
+	if err != nil {
+		stderr("fly: cannot open store: %v", err)
+		return 1
+	}
+
+	config, err := getConfig()
+	if err != nil {
+		stderr("fly: cannot get configuration: %v", err)
+		return 1
+	}
+
+	fn := &finder{
+		imageActionData: imageActionData{
+			s:                  s,
+			headers:            config.AuthPerHost,
+			dockerAuth:         config.DockerCredentialsPerRegistry,
+			insecureSkipVerify: globalFlags.InsecureSkipVerify,
+			debug:              globalFlags.Debug,
+		},
+		local:    flagLocal,
+		withDeps: true,
+	}
+
+	fn.ks = getKeystore()
+	if err := fn.findImages(&rktApps); err != nil {
+		stderr("%v", err)
+		return 1
+	}
+
+	u, err := types.NewUUID(uuid.New())
+	if err != nil {
+		stderr("fly: error creating UUID: %v", err)
+		return 1
+	}
+	dir := filepath.Join(flightDir(), u.String())
+	// TODO(jonboulle): lock this directory?
+	// TODO(jonboulle): require parent dir to exist?
+	err = os.MkdirAll(dir, 0700)
+	if err != nil {
+		stderr("fly: error creating directory: %v", err)
+		return 1
+	}
+
+	app := rktApps.Last()
+	id := app.ImageID
+	im, err := s.GetImageManifest(id.String())
+	if err != nil {
+		os.RemoveAll(dir)
+		stderr("fly: error getting image manifest: %v", err)
+		return 1
+	}
+	if im.App == nil {
+		os.RemoveAll(dir)
+		stderr("fly: image has no App section")
+		return 1
+	}
+	execargs := append(im.App.Exec, app.Args...)
+
+	//TODO(jonboulle): support overlay?
+	err = aci.RenderACIWithImageID(id, dir, s)
+	if err != nil {
+		os.RemoveAll(dir)
+		stderr("fly: error rendering ACI: %v", err)
+		return 1
+	}
+	// TODO(jonboulle): split this out
+	rfs := filepath.Join(dir, "rootfs")
+	execPath := filepath.Join(rfs, execargs[0])
+	if _, err := os.Stat(execPath); err != nil {
```

Comment:
In the case that execpath is a symlink, this check will fail. Maybe change order to chroot first, then stat execargs[0]